### DISCUSSION TITLE
External NPM library

### DISCUSSION THREAD

**Discussion by believethehype** [zauberzeug/nicegui#2284] [Q&A]
### Question

I couldn't find an example so I'm asking here. Is it possible to embed a npm package containing an external js library when calling js code from nicegui? I found the leaflet example that adds a .js lib to the header. When I apply import from "@lib" I currently get a Uncaught TypeError: Failed to resolve module specifier "@...". Relative references must start with either "/", "./", or "../". Is there a correct way to do this, or is it possible at all?

**Comment by falkoschindler** [comment]
Hi @believethehype!

I don't think referring to external NPM libraries is possible at the moment. As far as I know, references starting with "@" require a compile step to download and include such libraries. This process isn't implemented (yet), so you have to download and include all external libraries yourself.

We'd love to improve the integration of third-party JS/NPM/Vue libraries (see https://github.com/zauberzeug/nicegui/discussions/1454). But right now it's still a bit inconvenient.

**Reply by believethehype** [reply]
Thanks for the clarification. Did anyone do this yet and maybe have a tutorial on how to embed a downloaded npm lib into nicegui? (I'm not too knowledgeable in JS, that's why I'm looking forward to using nicegui). I want to access the browser's "window" to add an external signer extension to authorize a user within the app.
